### PR TITLE
Implement a user registration form.

### PR DESCRIPTION
**Pull request by Oleh-Stefanovych**
Implementation of form for registering a new user. It has fields avatar, login, password and password confirmation. Before sending data, each entered value validates:

- the avatar must be a picture and have a size of less than 500 kB;
- the login must consist of Latin letters of lower or upper case, length from 6 to 16 characters inclusive;
- the password must consist of Latin letters, have at least one lowercase letter, at least one upper case letter and at least one digit, length from 6 to 16 characters inclusive;
- repeated password must match the previously entered password.

### CODE REVIEW

**Comment by Oleh-Stefanovych** [comment]
@dkalinin PTAL.

File: app/view/component/registration-form/index.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+export class RegistrationForm extends Component {
+
+    constructor(parentRoot, id) {
+        super(id);
+        this._initRender(parentRoot);
+    }
+
+    _render() {
+        console.log("render form");
```
Comment (dkalinin) [inline app/view/component/registration-form/index.js]: Let's not use console.log.

File: app/view/component/registration-form/index.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        const root = document.createElement('form');
+
+        this.avatarChooserComponent = new ImageChooser(root, "avatar");
+
+        this.loginInputComponent = new LabeledInputWithMessage(root, "login")
+            .setLabelText("Enter the login:")
+            .setPlaceholder("Login");
```
Comment (dkalinin) [inline app/view/component/registration-form/index.js]: Input will be rendered 3 times here.

File: app/view/component/button/index.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import {Component} from "../";
+
+export class Button extends Component {
+
+    constructor(parentRoot, id) {
+        super(id);
+        this._text = "Button";
```
[the remark this answers is not in the document]
Comment (Oleh-Stefanovych) [reply app/view/component/button/index.js]: It is default text on button.

**Pull request merged**